geth/whisper/request_historic_messages.go

```diff
@@ -0,0 +1,151 @@
+package whisper
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/binary"
+	"fmt"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/whisper/whisperv5"
+	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/rpc"
+	"time"
+)
+
+//RequestHistoricMessagesHandler returns rpc which send p2p requestMessagesRequest for expired messages.
+func RequestHistoricMessagesHandler(nodeManager common.NodeManager) (rpc.Handler, error) {
+	whisper, err := nodeManager.WhisperService()
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := nodeManager.Node()
+	if err != nil {
+		return nil, err
+	}
+
+	return func(ctx context.Context, args ...interface{}) (interface{}, error) {
+		r, err := parseArgs(args)
+		if err != nil {
+			return nil, err
+		}
+
+		peer, err := extractIdFromEnode(r.Enode)
+		if err != nil {
+			return nil, err
+		}
+
+		symkey, err := whisper.GetSymKey(r.SymkeyID)
+		if err != nil {
+			return nil, err
+		}
+		env, err := makeEnvelop(r, symkey, node.Server().PrivateKey)
+		if err != nil {
+			return nil, err
+		}
+
+		err = whisper.RequestHistoricMessages(peer, env)
+		if err != nil {
+			return nil, err
+		}
+
+		return true, nil
+	}, nil
+}
+
+type requestMessagesRequest struct {
+	TimeLow  uint32
+	TimeUp   uint32
+	Topic    whisperv5.TopicType
+	SymkeyID string
+	Enode    string
+}
+
+func parseArgs(args ...interface{}) (requestMessagesRequest, error) {
+	var (
+		r = requestMessagesRequest{
+			TimeLow: 0,
+			TimeUp:  uint32(time.Now().Unix()),
+		}
+	)
+
+	if len(args) != 1 {
+		return requestMessagesRequest{}, fmt.Errorf("Invalid number of args")
```

Comment:
Tell more what's wrong. `invalid number of arguments, expected 1`. Also, use lowercase: https://github.com/golang/go/wiki/CodeReviewComments#error-strings